docs/docs/topics/mutual-auth.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Mutual Authentication
+lang: en-US
+sidebarDepth: 1
+meta:
+  - name: keywords
+    content: pomerium identity-access-proxy mutual authentication jwt jwks mtls
+description: >-
+  This page describes the concept of mutual authentication and why it's important.
+---
+
+# Mutual Authentication: A Component of Zero-Trust
+
+Pomerium provides a good layer of security out of the box, but it's not (and can't be) configured for complete [zero trust] right out of the box. This page explains several methods of achieving mutual authentication, a big part of the zero-trust model, with practical examples.
+
+This is a long page that dives into several specific security practices that provide mutual authentication. You can use the table of contents below to narrow down to the specific tools you're interested in, or read the entire doc for a deeper understanding of how these tools work together to support strong infrastructure security.
+
+[[toc]]
+
+## What Is Mutual Authentication?
+
+The concept of mutual authentication is deceptively simple. It means that both sides of a connection can validate the identity of the other. The application of this concept however, can be varied and complex. Pomerium provides the features and capability to provide mutual authentication between itself and end users, as well, as between itself and upstream services, but configuring those external endpoints can vary depending on each service's features.
+
+## Pomerium Defaults
+
+> - **Security**: Moderate
+> - **Difficulty**: Easy
+> - **Encrypted**: Yes (from the end user to Pomerium. It's up to the service to provide a TLS endpoint for Pomerium to use.)
+> - **Mutual Authentication**: None
+
+Let's look at a basic installation of Pomerium on a local network, with a single downstream service. This service contains sensitive data that we only want the right people to access.
```

Comment:
```suggestion
Let's look at a basic installation of Pomerium on a local network, with a single upstream service. This service contains sensitive data that we only want the right people to access.
```

We should add a terminology page like envoy. It's so confusing.

https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/intro/terminology

Reply:
I can do that.

Reply:
@alexfornuto  should we track that effort separately? I'm sure there's other terminology that would fit the bill. Some that come to mind are:

- upstream / downstream
- identity provider
- secure enclave / TEE / security keys
- peremeter / zero trust
- least user privilege
- Namespaces
- mutual authentication
- authentication
- authorization


and so on. Even if they are just links to other resources and our own docs.

Reply:
Yea, I'm thinking of a separate page, linking the first instance of each term in a page to it with anchors. 

In a future iteration of docs, we can make reusable tooltips to define concepts. See [this example](https://pantheon.io/docs/style-guide#tooltips).